src/main/java/de/zalando/aruha/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -28,6 +29,7 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
             final int hashValue = partitionKeyFields.stream()
                     // The problem is that JSONObject doesn't override hashCode(). Therefore convert it to
                     // a string first and then use hashCode()
+                    .map(pkf -> EventCategory.DATA.equals(eventType.getCategory()) ? "data." + pkf : pkf)
```

Comment:
@antban  👍 done https://github.com/zalando/nakadi/pull/291/commits/054d421217dfbf4215ddf1b0fd5b447d2f675276

I would not invest too much effort into refactoring all this category thing because I think it would not bring too much value now. In the future, when we focus on improving schema registry, then I would think about some generalisations in order to remove these special cases.